client/src/app/site/base/base-filter.service/base-filter-list.service.ts

```diff
@@ -90,63 +90,74 @@ export abstract class BaseFilterListService<V extends BaseViewModel> implements
      * @returns true if there are defined filters (regardless of current state)
      */
     public get hasFilterOptions(): boolean {
-        return !!this.filterDefinitions && this.filterDefinitions.length > 0;
+        return !!this.filterListData.filterDefinitions && this.filterListData.filterDefinitions.length > 0;
     }
 
-    /**
-     * Stack OsFilters
-     */
-    private _filterStack: OsFilterIndicator<V>[] = [];
-
     /**
      * get stacked filters
      */
     public get filterStack(): OsFilterIndicator<V>[] {
-        return this._filterStack;
+        return this.filterListData.filterStack;
+    }
+
+    protected get storageKey(): string {
+        return this.filterListData.storageKey;
     }
 
     /**
      * The key to access stored valued
      */
-    protected abstract readonly storageKey: string;
+    // protected abstract readonly storageKey: string;
 
-    public constructor(private store: StorageService, private historyService: HistoryService) {}
+    public constructor(private store: StorageService) {
+        this.filterListData = new FilterListData<V>();
```

Comment:
The idea behind the creating of an instance of `FilterListData` was not to just use it as data wrapper. Instead, please implement a real `create` or `build` function, which returns an instance of `FilterListData`. The `FilterListData` instance should then provide every function, which works on its data. The `BaseFilterListService` should provide functions, which filters a source by given `FilterDefinition`s.